### DISCUSSION TITLE
How to compile Rosetta 3.14 with Pytorch?

### DISCUSSION THREAD

**Discussion by zenzenith11** [RosettaCommons/rosetta#129] [Build & Install]
I am trying to compile Rosetta 3.14 with pytorch_gpu option on Ubuntu OS. Since 3.14 version supports extra=pytorch or pytorch gpu, I installed a Pytorch using Conda, and now have Pytorch2.4 under CUDA 11.8 installed. I am not sure if it is compatible with Rosetta and not sure how to compile?

I googled a lot but most methods seem to be out-dated (like below).
https://docs.rosettacommons.org/docs/latest/build_documentation/Building-Rosetta-with-TensorFlow-and-Torch
I can compile tensorflow, but not pytorch. 

Can aynone provide some guidance on this?

**Comment by lyskov** [comment]
Rosetta could be compiled with libtorch support (ie not with PyTorch support). As linked documentation mentioned all you need to do is to install libtorch and then compile Rosetta with `extras=torch` flag.

**Reply by zenzenith11** [reply]
> Rosetta could be compiled with libtorch support (ie not with PyTorch support). As linked documentation mentioned all you need to do is to install libtorch and then compile Rosetta with `extras=torch` flag.

3.14 version can be compiled with PyTorch as it is stated in the release note. But not sure how to do so.

**Reply by lyskov** [reply]
This is typo, what we really meant there is 'libTorch' support, - sorry for confusion!

**Comment by roccomoretti** [comment]
The page you link ([Building-Rosetta-with-TensorFlow-and-Torch](https://docs.rosettacommons.org/docs/latest/build_documentation/Building-Rosetta-with-TensorFlow-and-Torch)) should be up to date.

As Sergey mentions, Rosetta actually uses the C++ level `libTorch` library, rather than the Python level PyTorch library. As such, installing Pytorch with Conda doesn't actually help -- Rosetta won't be able to use that: you need to install the libtorch library files into the Rosetta directory prior to running the compilation, as mentioned on that page.

We did recognize that this is potentially confusing, which is why we changed the `extras` compilation option from `extras=pytorch` to `extras=torch`. However, I think the Rosetta 3.14 release is still working with the older `extras=pytorch` command line option.

If you're still encountering issues, posting the exact text of the error message you're getting during compilation (or during running if compilation looks to complete successfully) can help us better debug what the underlying issue may be.
